README.md

```diff
@@ -1,121 +1,281 @@
+during an update.
+
+Cmd | R/W | Data | Comment
+----|-----|------|---------------------
+0x80 | W  | 0x01 | 1st packet, 1st byte
+0x80 | W  | 0x02 | 1st packet, 2nd byte
+0x80 | W  | 0x03 | 1st packet, 3rd byte
+0x80 | W  | 0x04 | 1st packet, 4th byte
+0x80 | W  | 0x05 | 1st packet, 5th byte
+0x80 | W  | 0x06 | 1st packet, 6th byte
+0x80 | W  | 0x07 | 1st packet, 7th byte
+0x80 | W  | 0x08 | 1st packet, 8th byte
+0x80 | W  | 0xdc | 1st packet, checksum
+0x81 | R  | 0x01 | Commit 1st packet, OK response
+...
+0x80 | W  | 0x09 | nth packet, 1st byte
+0x80 | W  | 0x0a | nth packet, 2nd byte
+0x80 | W  | 0x0b | nth packet, 3rd byte
+0x80 | W  | 0x0c | nth packet, 4th byte
+0x80 | W  | 0x0d | nth packet, 5th byte
+0x80 | W  | 0x0e | nth packet, 6th byte
+0x80 | W  | 0x0f | nth packet, 7th byte
+0x80 | W  | 0x10 | nth packet, 8th byte
+0x80 | W  | 0x9c | nth packet, checksum
+0x81 | R  | 0x01 | Commit nth packet, OK response
+0x82 | W  | Any  | Reboot
+
+
+## Fail-Safe (Bootloader v3)
+
+Bootloader v3 introduces a fail-safe mechanism that allows entering 
+update mode by holding the reset button on power-up, even if the 
+main firmware is corrupted.
+
+The fail-safe mechanism requires that you have a system fully upgraded
+to bootloader v3. This can be done as follows:
+
+- When updating the SMC with an external programmer, the fail-safe is
+enabled if you use a release including bootloader v3.
+
+- When updating the SMC from the Commander X16, you must first update
+the bootloader to v3 and after that you must also update the firmware. The
+fail-safe mechanism is not enabled if you only update the bootloader.
+
+The fail-safe means that the update procedure can
+be started even if the firmware is bricked, for instance due to an
+aborted or failed update.
+
+It works as follows:
+
+- If the update procedure is interrupted during the firmware erase stage: Firmware
 erase starts from the last page. If interrupted during this stage,
-the reset vector at address 0x0000 is still unchanged, and a
-recovery update is possible.
+the reset vector at address 0x0000 is still unchanged and will be pointing
+to the bootloader's main entry point at address 0x1e00.
 
-- Update process interrupted after the whole firmware has been erased but
-before writing any parts of the new firmware to flash memory: When
-erasing the flash memory all words are set to byte value 0xffff, 
+- If the update procedure is interrupted after the whole firmware has been erased but
+before writing any part of the new firmware to flash memory: When
+erasing the flash memory, all words are set to byte value 0xffff, 
 which is interpreted as No Operation (NOP) by the SMC hardware. Execution
 starts from the reset vector at 0x0000 and continues until
-the first non NOP instruction at 0x1e00, the bootloader main function.
-This makes it possible to start the recovery update in this situation.
+the first non-NOP instruction at address 0x1e00, the bootloader's main entryp point.
```

Comment:
I just read through the updated PR. I did not notice any particular issue, except this small typo (entryp).